src/components/App/App.tsx

```diff
@@ -999,60 +1024,63 @@ export default class App extends React.Component<null, AppState> {
   render() {
     const sharer = this.state.participants.find((p) => p.isScreenShare);
     const controller = this.state.participants.find((p) => p.isController);
+
     return (
-      <React.Fragment>
-        {this.state.multiStreamSelection && (
-          <MultiStreamModal
-            streams={this.state.multiStreamSelection}
-            setMedia={this.setMedia}
-            resetMultiSelect={this.resetMultiSelect}
-          />
-        )}
-        {this.state.error && (
-          <Modal inverted basic open>
-            <Header as="h1" style={{ textAlign: 'center' }}>
-              {this.state.error}
-            </Header>
-            <div style={{ display: 'flex', justifyContent: 'center' }}>
-              <Button
-                primary
-                size="huge"
-                onClick={() => {
-                  window.location.href = '/';
-                }}
-                icon
-                labelPosition="left"
-              >
-                <Icon name="home" />
-                Go to home page
-              </Button>
-            </div>
-          </Modal>
-        )}
-        {!this.state.error && !this.state.isAutoPlayable && (
-          <Modal inverted basic open>
-            <div style={{ display: 'flex', justifyContent: 'center' }}>
-              <Button
-                primary
-                size="huge"
-                onClick={() => {
-                  this.setState({ isAutoPlayable: true });
-                  this.setMute(false);
-                }}
-                icon
-                labelPosition="left"
-              >
-                <Icon name="sign-in" />
-                Join Party
-              </Button>
-            </div>
-          </Modal>
-        )}
-        <TopBar user={this.state.user} />
-        {
-          <Grid stackable celled="internally">
-            <Grid.Row>
-              <Grid.Column width={12} className="fullHeightColumn">
+      <LobbyWrapper io={this.socket}>
```

Comment:
I'm not entirely sure what the purpose of these Wrapper classes is.

I assume they're providing state to the components within them, but this can also be achieved by passing state and necessary mutation functions down as props. At some point we'll probably want to add some kind of state management library (Redux, etc.), and then we'll have a different way of sharing state anyway.

Maybe we can start with the props solution for this change and refactor later?